src/exercise/DiscountExercise.sol

```diff
@@ -133,31 +165,90 @@ contract DiscountExercise is BaseExercise {
+    }
+
+    /// @notice Sets the discount minAmountToTriggerSwap.
+    /// @param _minAmountToTriggerSwap The new minAmountToTriggerSwap
+    function setMinAmountToTriggerSwap(uint256 _minAmountToTriggerSwap) external onlyOwner {
+        _setMinAmountToTriggerSwap(_minAmountToTriggerSwap);
+    }
+
+    function _setMinAmountToTriggerSwap(uint256 _minAmountToTriggerSwap) internal {
+        minAmountToTriggerSwap = _minAmountToTriggerSwap;
+        emit SetMinAmountToTrigger(_minAmountToTriggerSwap);
+    }
+
+    function pause() external onlyOwner {
+        _pause();
+    }
+
+    function unpause() external onlyOwner {
+        _unpause();
+    }
 
-    function _exercise(address from, uint256 amount, address recipient, bytes memory params)
+    /// Internal functions
+    function _zap(address from, uint256 amount, address recipient, DiscountExerciseParams memory params)
         internal
         virtual
         returns (uint256 paymentAmount, address, uint256, uint256)
     {
-        // decode params
-        DiscountExerciseParams memory _params = abi.decode(params, (DiscountExerciseParams));
-
-        if (block.timestamp > _params.deadline) revert Exercise__PastDeadline();
+        if (block.timestamp > params.deadline) revert Exercise__PastDeadline();
+        uint256 discountedUnderlying = amount.mulDivUp(multiplier, BPS_DENOM);
+        uint256 fee = discountedUnderlying.mulDivUp(instantExitFee, BPS_DENOM);
+        uint256 underlyingAmount = discountedUnderlying - fee;
+
+        // Fee amount in underlying tokens charged for zapping
+        feeAmount += fee;
+
+        if (feeAmount >= minAmountToTriggerSwap) {
+            uint256 minAmountOut = _getMinAmountOutData(feeAmount, swapProps.maxSwapSlippage, address(oracle));
+            /* Approve the underlying token to make swap */
+            underlyingToken.approve(swapProps.swapper, feeAmount);
+            /* Swap underlying token to payment token (asset) */
+            _generalSwap(swapProps.exchangeTypes, address(underlyingToken), address(paymentToken), feeAmount, minAmountOut, swapProps.exchangeAddress);
+            feeAmount = 0;
+            // transfer payment tokens from user to the set receivers
+            distributeFees(paymentToken.balanceOf(address(this)), paymentToken);
+            if (paymentToken.balanceOf(feeRecipients[0]) == 0 || paymentToken.balanceOf(feeRecipients[1]) == 0) {
+                revert Exercise__FeeDistributionFailed();
+            }
+        }
 
-        // apply multiplier to price
-        uint256 price = oracle.getPrice().mulDivUp(multiplier, MULTIPLIER_DENOM);
+        // transfer underlying tokens to recipient without the bonus
+        _pay(recipient, underlyingAmount);
 
-        paymentAmount = amount.mulWadUp(price);
-        if (paymentAmount > _params.maxPaymentAmount) revert Exercise__SlippageTooHigh();
+        emit Exercised(from, recipient, underlyingAmount, paymentAmount);
```

Comment:
this is prob fine but it would be nice to know what kind of exercise zap or redeem this was in the event.  however... its not worth doing anything about...

enum exerciseType maybe